ddev/src/ddev/utils/scripts/check_pr.py

```diff
@@ -0,0 +1,189 @@
+"""
+Running this script by itself must not use any external dependencies.
+"""
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import annotations
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+from typing import Iterator
+
+
+def requires_changelog(target: str, files: Iterator[str]) -> bool:
+    if target == 'ddev':
+        source = 'src/ddev/'
+    else:
+        if target == 'datadog_checks_base':
+            directory = 'base'
+        elif target == 'datadog_checks_dev':
+            directory = 'dev'
+        elif target == 'datadog_checks_downloader':
+            directory = 'downloader'
+        else:
+            directory = target.replace('-', '_')
+
+        source = f'datadog_checks/{directory}/'
+
+    return any(f.startswith(source) or f == 'pyproject.toml' for f in files)
+
+
+def git(*args) -> str:
+    try:
+        process = subprocess.run(
+            ['git', *args], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, encoding='utf-8', check=True
+        )
+    except subprocess.CalledProcessError as e:
+        raise OSError(f'{str(e)[:-1]}:\n{e.output}') from None
```

Comment:
Are we changing the exception type to work nicely with github actions somehow?

Reply:
right, so we're wrapping essentially. Any particular reason `OSError` and not `RuntimeError` (or custom exception, though seems like overkill for this script). Also, what about `raise from`?

Reply:
Will switch to `runtimeerror`, won't do `raise from`. The problem with `raise from` is that it produces 2 tracebacks.  With the current setup, we get both exception and the command output next to each other.